### ISSUE TITLE
`ctrl-p` quick open: suggestions from the repository of currently active file should be prioritized more in case of multiple matches with same file name

### ISSUE THREAD

**Issue by jakub-g** [microsoft/vscode#77643]
When you launch quick open (`ctrl-p`) and type some text (`filename.txt`), the order of suggestions will be as follows:
- _recently opened_ matches
- other matches

How the sorting happens _inside those groups_ is a bit unclear, especially in case of matches with same score (for example, dozens of files with the same filename).

Anyway, the behavior is suboptimal for common file names (like `package.json` or `README.md`) when you have a workspace with a dozen repositories opened at the same time.

I think the order should also take into consideration the _currently active file_.

---

Consider following workspace with repositories `r1`, `r2`, .., `r10` in it.

Let's say I'm working on `r3/src/foobar.js`. When I type `ctrl-p`, `package.json`, I'm very likely interested in `r3/package.json`. However vscode will suggest me the `package.json` that was the most recently used, for example `r7/package.json`. It could be that `r3/package.json` will be on the very bottom of the list.

_Aside: typing `r3 package.json` will result in 0 matches - the names of repositories are not included in the searching process._

TL;DR: In case of multiple matches with same score, the one from _the same repository as the currently active file in editor_ should be the highest ranked.

**Comment by roblourens** [comment]
/duplicate of https://github.com/microsoft/vscode/issues/5806
